pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test: TestDataDriven failed

pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test.TestDataDriven [failed](https://tanzanite.cluster.engflow.com/invocations/default/a6d38f79-7ebf-46b9-85c1-70f7b68563f3?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NwYW5jb25maWdjY2wvc3BhbmNvbmZpZ3JlY29uY2lsZXJjY2w6c3BhbmNvbmZpZ3JlY29uY2lsZXJjY2xfdGVzdA==) on master @ [ef87a4e4816fc6a208b78307221ec56667123877](https://github.com/cockroachdb/cockroach/commits/ef87a4e4816fc6a208b78307221ec56667123877):


```
        /Tenant/10/Table/4{7-8}                    ignore_strict_gc=true rangefeed_enabled=true exclude_data_from_backup=true
        /Tenant/10/Table/4{8-9}                    database system (tenant)
        /Tenant/10/Table/5{0-1}                    database system (tenant)
        /Tenant/10/Table/5{1-2}                    database system (tenant)
        /Tenant/10/Table/5{2-3}                    database system (tenant)
        /Tenant/10/Table/5{3-4}                    database system (tenant)
        /Tenant/10/Table/5{4-5}                    database system (tenant)
        /Tenant/10/Table/5{5-6}                    database system (tenant)
        /Tenant/10/Table/5{6-7}                    database system (tenant)
        /Tenant/10/Table/5{7-8}                    database system (tenant)
        /Tenant/10/Table/5{8-9}                    database system (tenant)
        /Tenant/10/Table/{59-60}                   database system (tenant)
        /Tenant/10/Table/6{0-1}                    database system (tenant)
        /Tenant/10/Table/6{1-2}                    database system (tenant)
        /Tenant/10/Table/6{2-3}                    database system (tenant)
        /Tenant/10/Table/6{3-4}                    database system (tenant)
        /Tenant/10/Table/6{4-5}                    database system (tenant)
        /Tenant/10/Table/6{5-6}                    database system (tenant)
        /Tenant/10/Table/6{6-7}                    database system (tenant)
        /Tenant/11{-\x00}                          database system (tenant)
        /Tenant/12{-\x00}                          database system (tenant)
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/multitenant/basic:195:
        exec-sql [1 args]
        CREATE DATABASE db;
        CREATE TABLE db.t1();
        CREATE TABLE db.t2();
        CREATE SCHEMA db.sc;
        CREATE TYPE typ AS ENUM();
        CREATE VIEW v AS SELECT 1;
        CREATE SEQUENCE db.seq;
        CREATE MATERIALIZED VIEW mv AS SELECT 1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/multitenant/basic:206:
         
        expected:
        upsert /Tenant/10/Table/10{6-7}            range default
        upsert /Tenant/10/Table/10{7-8}            range default
        upsert /Tenant/10/Table/11{2-3}            range default
        upsert /Tenant/10/Table/11{3-4}            range default
        
        found:
        upsert /Tenant/10/Table/11{6-7}            range default
        upsert /Tenant/10/Table/11{7-8}            range default
        upsert /Tenant/10/Table/12{2-3}            range default
        upsert /Tenant/10/Table/12{3-4}            range default
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestDataDriven_multitenant_basic3770369990
        --- FAIL: TestDataDriven/multitenant/basic (24.57s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38361

pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test.TestDataDriven [failed](https://tanzanite.cluster.engflow.com/invocations/default/5d634410-f43b-423c-98ad-984c82ccf30e?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NwYW5jb25maWdjY2wvc3BhbmNvbmZpZ3JlY29uY2lsZXJjY2w6c3BhbmNvbmZpZ3JlY29uY2lsZXJjY2xfdGVzdA==) on master @ [c511a3284cd821c40145f151537dd6b971dce1eb](https://github.com/cockroachdb/cockroach/commits/c511a3284cd821c40145f151537dd6b971dce1eb):


```
        /Tenant/10/Table/4{7-8}                    ignore_strict_gc=true rangefeed_enabled=true exclude_data_from_backup=true
        /Tenant/10/Table/4{8-9}                    database system (tenant)
        /Tenant/10/Table/5{0-1}                    database system (tenant)
        /Tenant/10/Table/5{1-2}                    database system (tenant)
        /Tenant/10/Table/5{2-3}                    database system (tenant)
        /Tenant/10/Table/5{3-4}                    database system (tenant)
        /Tenant/10/Table/5{4-5}                    database system (tenant)
        /Tenant/10/Table/5{5-6}                    database system (tenant)
        /Tenant/10/Table/5{6-7}                    database system (tenant)
        /Tenant/10/Table/5{7-8}                    database system (tenant)
        /Tenant/10/Table/5{8-9}                    database system (tenant)
        /Tenant/10/Table/{59-60}                   database system (tenant)
        /Tenant/10/Table/6{0-1}                    database system (tenant)
        /Tenant/10/Table/6{1-2}                    database system (tenant)
        /Tenant/10/Table/6{2-3}                    database system (tenant)
        /Tenant/10/Table/6{3-4}                    database system (tenant)
        /Tenant/10/Table/6{4-5}                    database system (tenant)
        /Tenant/10/Table/6{5-6}                    database system (tenant)
        /Tenant/10/Table/6{6-7}                    database system (tenant)
        /Tenant/11{-\x00}                          database system (tenant)
        /Tenant/12{-\x00}                          database system (tenant)
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/multitenant/basic:195:
        exec-sql [1 args]
        CREATE DATABASE db;
        CREATE TABLE db.t1();
        CREATE TABLE db.t2();
        CREATE SCHEMA db.sc;
        CREATE TYPE typ AS ENUM();
        CREATE VIEW v AS SELECT 1;
        CREATE SEQUENCE db.seq;
        CREATE MATERIALIZED VIEW mv AS SELECT 1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/multitenant/basic:206:
         
        expected:
        upsert /Tenant/10/Table/10{6-7}            range default
        upsert /Tenant/10/Table/10{7-8}            range default
        upsert /Tenant/10/Table/11{2-3}            range default
        upsert /Tenant/10/Table/11{3-4}            range default
        
        found:
        upsert /Tenant/10/Table/11{6-7}            range default
        upsert /Tenant/10/Table/11{7-8}            range default
        upsert /Tenant/10/Table/12{2-3}            range default
        upsert /Tenant/10/Table/12{3-4}            range default
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestDataDriven_multitenant_basic921388296
        --- FAIL: TestDataDriven/multitenant/basic (41.03s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/BUILD.bazel
+++ b/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/BUILD.bazel
@@ -25,6 +25,7 @@ go_test(
         "//pkg/testutils",
         "//pkg/testutils/datapathutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/hlc",

--- a/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/datadriven_test.go
+++ b/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/datadriven_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/sqllivenesstestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -85,7 +86,8 @@ import (
 // must be cleared out.
 func TestDataDriven(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-
+	skip.UnderRace(t, "descriptor ID generation is not deterministic under race")
+	skip.UnderDeadlock(t, "descriptor ID generation is not deterministic under deadlock")
 	ctx := context.Background()
 	datadriven.Walk(t, datapathutils.TestDataPath(t), func(t *testing.T, path string) {
 		defer log.Scope(t).Close(t)

